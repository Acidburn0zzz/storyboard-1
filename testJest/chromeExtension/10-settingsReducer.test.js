/* eslint-env jest */
import reducer from '../../lib/chromeExtension/reducers/settingsReducer';

describe('settingsReducer', () => {
  it('should update settings with new values', () => {
    let state = reducer(undefined, { type: '' });
    expect(state).toMatchSnapshot();
    state = reducer(state, {
      type: 'UPDATE_SETTINGS',
      settings: {
        maxRecords: 2e3,
        forgetHysteresis: 0.5,
      },
    });
    expect(state).toMatchSnapshot();
  });

  it('should not allow invalid maxRecords', () => {
    let state = reducer(undefined, { type: '' });
    state = reducer(state, {
      type: 'UPDATE_SETTINGS',
      settings: {
        maxRecords: -25,
      },
    });
    expect(state.maxRecords >= 0).toBeTruthy();
  });
});
