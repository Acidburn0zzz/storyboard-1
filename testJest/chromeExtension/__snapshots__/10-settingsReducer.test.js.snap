exports[`settingsReducer should update settings with new values 1`] = `
Object {
  "colorClientBg": "aliceblue",
  "colorClientBgIsDark": false,
  "colorClientFg": "black",
  "colorServerBg": "rgb(214, 236, 255)",
  "colorServerBgIsDark": false,
  "colorServerFg": "black",
  "colorUiBg": "white",
  "colorUiBgIsDark": false,
  "colorUiFg": "rgb(64, 64, 64)",
  "fCollapseAllNewStories": false,
  "fDiscardRemoteClientLogs": false,
  "fExpandAllNewAttachments": false,
  "fShorthandForDuplicates": true,
  "fShowClosedActions": false,
  "forgetHysteresis": 0.25,
  "maxRecords": 800,
  "timeType": "LOCAL",
}
`;

exports[`settingsReducer should update settings with new values 2`] = `
Object {
  "colorClientBg": "aliceblue",
  "colorClientBgIsDark": false,
  "colorClientFg": "black",
  "colorServerBg": "rgb(214, 236, 255)",
  "colorServerBgIsDark": false,
  "colorServerFg": "black",
  "colorUiBg": "white",
  "colorUiBgIsDark": false,
  "colorUiFg": "rgb(64, 64, 64)",
  "fCollapseAllNewStories": false,
  "fDiscardRemoteClientLogs": false,
  "fExpandAllNewAttachments": false,
  "fShorthandForDuplicates": true,
  "fShowClosedActions": false,
  "forgetHysteresis": 0.5,
  "maxRecords": 2000,
  "timeType": "LOCAL",
}
`;
